satpy/readers/__init__.py

```diff
@@ -391,6 +391,9 @@ def load_readers(filenames=None, reader=None, reader_kwargs=None,
     if not filenames and not reader:
         # used for an empty Scene
         return {}
+    elif reader and filenames is not None and not filenames:
+        # user made a mistake in their glob pattern
+        raise ValueError("'filenames' was provided but is empty.")
```

Comment:
Can this be strange if one use `find_files_and_readers` ?

Reply:
No because reader should be `None` and filenames will be a dictionary. If filenames is an empty dictionary (I think `find_files_and_readers` raises an exception if no files are found) then we would want the same error: "You provided a dictionary of readers and filenames, but with no readers or filenames". This does not catch the case of someone providing `{"reader_name": []}` but maybe we don't need to handle that case?

Reply:
ok, fine